flow360/component/simulation/models/surface_models.py

```diff
@@ -0,0 +1,130 @@
+"""
+Contains basically only boundary conditons for now. In future we can add new models like 2D equations.
+"""
+
+from abc import ABCMeta
+from typing import List, Literal, Optional, Tuple, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.operating_condition import VelocityVectorType
+
+from .turbulence_quantities import TurbulenceQuantitiesType
+
+
+class BoundaryBase(Flow360BaseModel, metaclass=ABCMeta):
+    type: str = pd.Field()
+    entities: EntityList[Surface] = pd.Field(alias="surfaces")
+
+
+class BoundaryBaseWithTurbulenceQuantities(BoundaryBase, metaclass=ABCMeta):
+    turbulence_quantities: Optional[TurbulenceQuantitiesType] = pd.Field(None)
+
+
+class HeatFlux(Flow360BaseModel):
+    heat_flux: Union[HeatFluxType, pd.StrictStr] = pd.Field()
+
+
+class Temperature(Flow360BaseModel):
+    temperature: Union[TemperatureType, pd.StrictStr] = pd.Field()
+
+
+class TotalPressure(Flow360BaseModel):
+    total_pressure: PressureType = pd.Field()
+
+
+class Pressure(Flow360BaseModel):
+    pressure: PressureType = pd.Field()
+
+
+class MassFlowRate(Flow360BaseModel):
+    mass_flow_rate: MassFluxType.NonNegative = pd.Field()
+
+
+class Mach(Flow360BaseModel):
+    Mach: NonNegativeFloat = pd.Field()
+
+
+class Wall(BoundaryBase):
+    """Replace Flow360Param:
+    - NoSlipWall
+    - IsothermalWall
+    - HeatFluxWall
+    - WallFunction
+    - SolidIsothermalWall
+    - SolidAdiabaticWall
+    """
+
+    type: Literal["Wall"] = pd.Field("Wall", frozen=True)
+    use_wall_function: bool = pd.Field(False)
+    velocity: Optional[VelocityVectorType] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+    heat_spec: Optional[Union[HeatFluxType, TemperatureType, HeatFlux, Temperature]] = pd.Field(
+        None
+    )
+
+
+class Freestream(BoundaryBaseWithTurbulenceQuantities):
+    type: Literal["Freestream"] = pd.Field("Freestream", frozen=True)
+    velocity: Optional[VelocityVectorType] = pd.Field(None)
+    velocity_type: Literal["absolute", "relative"] = pd.Field("relative")
+
+
+class Outflow(BoundaryBaseWithTurbulenceQuantities):
+    """Replace Flow360Param:
+    - SubsonicOutflowPressure
+    - SubsonicOutflowMach
+    - MassOutflow
+    """
+
+    type: Literal["Outflow"] = pd.Field("Outflow", frozen=True)
+    spec: Union[PressureType, MassFluxType.NonNegative, Pressure, MassFlowRate, Mach] = pd.Field()
```

Comment:
Same here. The `PressureType `is included in `Pressure `right?

Reply:
Note: We will implement auto type detection. and convert input to Class(value = input)